azure-iot-device/tests/iothub/pipeline/test_pipeline_stages_iothub_mqtt.py

```diff
@@ -123,6 +134,51 @@ def apply_fake_pipeline_thread(fake_pipeline_thread):
 )
 
 
+def create_message_with_user_properties(message_content, is_multiple):
+    m = Message(message_content)
+    m.custom_properties[fake_message_user_property_1_key] = fake_message_user_property_1_value
+    if is_multiple:
+        m.custom_properties[fake_message_user_property_2_key] = fake_message_user_property_2_value
+    return m
+
+
+def create_message_with_system_and_user_properties(message_content, is_multiple):
+    if is_multiple:
+        msg = Message(message_content, message_id=fake_message_id, content_type=fake_content_type)
+    else:
+        msg = Message(message_content, message_id=fake_message_id)
+
+    msg.custom_properties[fake_message_user_property_1_key] = fake_message_user_property_1_value
+    if is_multiple:
+        msg.custom_properties[fake_message_user_property_2_key] = fake_message_user_property_2_value
+    return msg
+
+
+def create_message_for_output_with_user_properties(message_content, is_multiple):
+    m = Message(message_content, output_name=fake_output_name)
+    m.custom_properties[fake_message_user_property_1_key] = fake_message_user_property_1_value
+    if is_multiple:
+        m.custom_properties[fake_message_user_property_2_key] = fake_message_user_property_2_value
+    return m
+
+
+def create_message_for_output_with_system_and_user_properties(message_content, is_multiple):
+    if is_multiple:
+        msg = Message(
+            message_content,
+            output_name=fake_output_name,
+            message_id=fake_message_id,
+            content_type=fake_content_type,
+        )
+    else:
+        msg = Message(message_content, output_name=fake_output_name, message_id=fake_message_id)
+
+    msg.custom_properties[fake_message_user_property_1_key] = fake_message_user_property_1_value
+    if is_multiple:
+        msg.custom_properties[fake_message_user_property_2_key] = fake_message_user_property_2_value
+    return msg
+
```

Comment:
It would probably be better for these to be handled as a fixture. But I suspect that's likely going to be too difficult to do with the current model of testing employed in this module.